programs/vote/src/vote_state/mod.rs

```diff
@@ -226,6 +279,172 @@ impl VoteStateUpdate {
     }
 }
 
+/// Ignoring overhead, in a full `VoteStateUpdate` the lockouts take up
+/// 31 * (64 + 32) = 2976 bits.
+///
+/// In this schema we separate the votes into 3 separate lockout structures
+/// and store offsets rather than slot number, allowing us to use smaller fields.
+///
+/// In a full `CompactVoteStateUpdate` the lockouts take up
+/// (64 + 8) + (32 + 8) * 16 + (16 + 8) * 8 + (8 + 8) * 6 = 1000 bits
+/// allowing us to greatly reduce block size.
+#[frozen_abi(digest = "BfeSJNsfQeX6JU7dmezv1s1aSvR5SoyxKRRZ4ubTh2mt")]
+#[derive(Serialize, Default, Deserialize, Debug, PartialEq, Eq, Clone, AbiExample)]
+pub struct CompactVoteStateUpdate {
+    /// The proposed root
+    pub root: Option<Slot>,
+    /// The first (bottom) vote in tower
+    pub first: CompactLockout<u64>,
+
+    /// Part of the proposed tower, votes with confirmation_count >= 15
+    pub lockouts_32: VecDeque<CompactLockout<u32>>,
+    /// Part of the proposed tower, votes with 15 > confirmation_count >= 7
+    pub lockouts_16: VecDeque<CompactLockout<u16>>,
+    /// Part of the proposed tower, votes with 7 > confirmation_count
+    pub lockouts_8: VecDeque<CompactLockout<u8>>,
```

Comment:
We could use a `ShortVec` here to save an additional 21 bytes in each vote instruction

Reply:
a u8 might not be sufficient here right?

If you had votes `0 1 2 3 4 5 6 7 8 300`, the tower would look like `0 300` and not representable by a u8?

Reply:
ah yeah, I think that for this system to work I need to keep the offset to the next block rather than from the previous block